src/porepy/fracs/wells_3d.py

```diff
@@ -163,108 +233,123 @@ def __init__(
         parameters: Optional[dict] = None,
     ) -> None:
 
-        self.well_dim = 1
-        self.wells = wells if wells is not None else []
+        self.well_dim: int = 1
+        """All polyline wells have dimension 1."""
+        self.wells: list[Well] = wells if wells is not None else []
         """List of wells in the network."""
 
         for i, w in enumerate(self.wells):
             w.index = i
 
-        if parameters is None:
-            self.parameters = {}
-            """Dictionary of parameters, e.g. for the meshing process."""
-        else:
+        self.parameters: dict
+        """Dictionary of parameters, e.g. for the meshing process passed at
+        instantiation."""
+
+        if parameters is not None:
```

Comment:
Was the change here on purpose? If parameters is not specified, it is now None, but it used to be an empty dict.

Reply:
@vlipovac: Can you please handle this one?

Reply:
My bad, I fixed it by defining `self.parameters` as an empty dict before checking the input arg and assigning.

There is one more todo though, as I have noted in the class docstring (regarding the `domain` attribute)

Reply:
The only usage of `self.domain` is in line 442:

``` python3
# Finally, update tags for the well's faces (boundary, tip, fracture).
bounding_planes = self.domain.polytope_from_bounding_box()
boundary = np.zeros(2, dtype=bool)
```

This suggests that ``domain`` cannot be ``None``.  Since this is "your" code @IvarStefansson. What should we proceed?

Reply:
It seems to me that the safest option is to make domain mandatory

Reply:
Yes. Thanks. I will take care of it.